### DISCUSSION TITLE
アイコンに関するライブラリ選定について

### DISCUSSION THREAD

**Discussion by kagomen** [first-contributions-ja/first-contributions-ja.github.io#86] [10: 技術についての質問] [answered]
## アイコン類をどうするか？
現状、ABOUTセクションにてGitHubのロゴしか使用予定がないのですが、

- 今後メニューなどが増えたときに必要になるかも？
- ライブラリを使ってみたい！

といった理由で、ライブラリの導入を検討しています。

## ライブラリたくさんありすぎて、何を選べばいいかわからない

Next.jsがサポートしてるライブラリは、`next.config.js`で設定すると良い感じに最適化してくれるらしい。

参考: [optimizePackageImports -Next.js](https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports)

なのでNext.jsがサポートしてるライブラリから選ぼうかな、と。

## 関連するイシュー
#56

**Comment by kagomen** [comment]
[optimizePackageImports -Next.js](https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports)
上記のページで、アイコンに関するライブラリと思われるものを抜粋

- lucide
- heroicons
- mui
- tabler
- react-icons

ですかね

**Comment by kagomen** [comment, marked as answer]
- lucide
  - アイコンの数1447種類
  - デザインに癖あり
  - <img width="81" alt="スクリーンショット 2024-03-20 20 28 17" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/c79f42e8-1260-4ffb-9d3a-3437657e1395">

- heroicons
  - tailwindのクラス指定が使える！
    - Tailwindの開発者が開発したらしい
  - アイコンの数288種類しかない
  - 肝心のGitHubのアイコンがなかった！→却下
 
- Material UI
  - Googleのマテリアルデザインをベースに開発されたReact向けのライブラリ
    - ずっとGoogleが開発してるんだと思ってた！
  - アイコンの種類2100以上
  - デザインの安定感
  - <img width="72" alt="スクリーンショット 2024-03-20 20 34 18" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/5b58de62-0a07-4992-8bec-271d52ebc52f">

- tabler
  - アイコンの数4564種類
  - デザインに癖あり（lucideと似てる、これが流行りなのか）
  - <img width="78" alt="スクリーンショット 2024-03-20 20 30 32" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/bc16c71e-93e2-4f02-9d04-2b517f4c8317">
 
- react-icons
  - [あらゆるアイコンがある](https://react-icons.github.io/react-icons/)
  - この中から選ぶのも大変……！

**Reply by kagomen** [reply]
気になった記事（時間あるときに読みたい）
[こんにちはMUI！ 新しくなったMaterial UI v5 -Zenn](https://zenn.dev/yumemi_inc/articles/2021-09-26-material-ui-v5)
[React向けUIコンポーネントライブラリ、Material UI(v4)の紹介 -Zenn](https://zenn.dev/yumemi_inc/articles/2020-09-24-react-material-ui?redirected=1)

**Comment by kazzyfrog** [comment]
他に使うとしたら、
ダークモードの月と太陽とか、X（Twitter）のアイコンもあったので、Material Icon良さそうですね！
